### PR TITLE
KAA-829 Fix all warnings

### PR DESCRIPTION
**Pull request by rasendubi**
This patch enables and fixes all warnings (-Wall -Wextra -pedantic).

Some targets (esp8266, I look at you) has warning in the SDK, so these
warning are disabled for them.

It also enables -Werror for Travis, so from this point on we keep our
code warning-free.

/cc @forGGe

### CODE REVIEW

File: client/client-multi/client-c/build.sh [hunk 1]
```diff
@@ -38,7 +38,7 @@ COLLECT_COVERAGE=0
 prepare_build() {
     mkdir -p build;
     cd build;
-    cmake -DKAA_DEBUG_ENABLED=$DEBUG_ENABLED -DKAA_MAX_LOG_LEVEL=$MAX_LOG_LEVEL -DKAA_UNITTESTS_COMPILE=$UNITTESTS_COMPILE -DKAA_COLLECT_COVERAGE=$COLLECT_COVERAGE ..;
+    cmake -DKAA_DEBUG_ENABLED=$DEBUG_ENABLED -DKAA_MAX_LOG_LEVEL=$MAX_LOG_LEVEL -DKAA_UNITTESTS_COMPILE=$UNITTESTS_COMPILE -DKAA_COLLECT_COVERAGE=$COLLECT_COVERAGE .. -DCMAKE_C_FLAGS="-Werror";
```
Comment (forGGe) [inline client/client-multi/client-c/build.sh:41]: Not critical comment: let's think about passing build type via CMAKE_BUILD_TYPE . Debug build may expose then  the -Werror flag somewhere in listfiles using CMAKE_C_FLAGS_DEBUG. This adds some indirection and ability to add more flags if needed for development purposes.

ATM this not critical, just a cause to have a chat.

File: client/client-multi/client-c/src/kaa/platform-impl/esp8266/esp8266_logger.c [hunk 2]
```diff
@@ -21,13 +21,18 @@
 
 #include "../../platform/ext_system_logger.h"
 
+int snprintf (char *str, size_t count, const char *fmt, ...);
+int vsnprintf (char *str, size_t count, const char *fmt, va_list arg);
```
Comment (forGGe) [inline client/client-multi/client-c/src/kaa/platform-impl/esp8266/esp8266_logger.c]: Space between fn_name and '('

File: client/client-multi/client-c/src/kaa/platform-impl/posix/posix_kaa_client.c [hunk 3]
```diff
@@ -71,7 +72,8 @@ static const int OPERATIONS_SERVICES_COUNT = sizeof(OPERATIONS_SERVICES) / sizeo
 #define MAX_LOG_COUNT           SIZE_MAX
 #define MAX_LOG_BUCKET_SIZE     (KAA_TCP_CHANNEL_OUT_BUFFER_SIZE >> 3)
 
-_Static_assert(MAX_LOG_BUCKET_SIZE, "Maximum bucket size cannot be 0!");
+// _Static_assert(MAX_LOG_BUCKET_SIZE, "Maximum bucket size cannot be 0!");
```
Comment (forGGe) [inline client/client-multi/client-c/src/kaa/platform-impl/posix/posix_kaa_client.c]: Delete commented code.

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c [hunk 4]
```diff
@@ -67,61 +69,67 @@ static void reset_access_point_test_info()
...
+
+static uint8_t CONNECT_HEAD[] = {0x35, 0x46};
+static uint8_t CONNECT_PACK[] = {0x34, 0x45};
+
+static uint8_t DESTINATION_SOCKADDR[]   = {
+    0x02, 0x00, 0x26, 0xa0, 0xc0, 0xa8, 0x4d, 0x02,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+
+static uint8_t CONNECTION_DATA[]   = {
```
Comment (forGGe) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c:91]: God, at last someone did it!

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c [hunk 5]
```diff
@@ -460,7 +468,7 @@ void test_set_access_point(kaa_transport_channel_interface_t *channel)
     //Use connection data to destination 192.168.77.2:9888
     kaa_access_point_t access_point;
     access_point.id = 10;
-    access_point.connection_data = CONNECTION_DATA;
+    access_point.connection_data = (char *)CONNECTION_DATA;
```
Comment (forGGe) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c:471]: Hope in some time in a future we'll use uint8_t for **all** byte buffers

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c [hunk 6]
```diff
@@ -526,9 +536,13 @@ kaa_error_t kaa_platform_protocol_process_server_sync(kaa_platform_protocol_t *s
                                                     , const char *buffer
                                                     , size_t buffer_size)
 {
+    (void)self;
+    // TODO(rasen): oh-oh. we must use buffer_size here
```
Comment (forGGe) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c:540]: Let's open ticket for it

File: client/client-multi/client-c/test/test_kaa_profile.c [hunk 7]
```diff
@@ -374,6 +387,9 @@ void test_profile_handle_sync(void)
     KAA_TRACE_OUT(logger);
 }
 
+// Mom, I'm sorry. Promise to fix that later
+extern kaa_error_t kaa_profile_force_sync(kaa_profile_manager_t *self);
```
Comment (forGGe) [inline client/client-multi/client-c/test/test_kaa_profile.c:391]: I know I'm bad person, but they forced me to write this ^^

**Comment by forGGe** [comment]
Minor nitpicks

File: client/client-multi/client-c/build.sh (hunk 1 above)
Comment (RostakaGmfun) [inline client/client-multi/client-c/build.sh:41]: KAA-858 addresses this concept

Comment (forGGe) [inline client/client-multi/client-c/build.sh:41]: What about merging KAA-858 into true_metal then?

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c (hunk 6 above)
Comment (rasendubi) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c:540]: That's a test, so I'm not sure

File: client/client-multi/client-c/src/kaa/platform-impl/cc32xx/logger.c [hunk 8]
```diff
@@ -24,6 +24,7 @@
 
 void ext_write_log(FILE * sink, const char * buffer, size_t message_size)
 {
+    (void)message_size;
     if (!buffer || !sink) {
```
Comment (RostakaGmfun) [inline client/client-multi/client-c/src/kaa/platform-impl/cc32xx/logger.c:28]: Should we check `sink` value, if we don't use it?

File: client/client-multi/client-c/build.sh (hunk 1 above)
Comment (rasendubi) [inline client/client-multi/client-c/build.sh:41]: Debug != developer build

File: client/client-multi/client-c/src/kaa/platform-impl/esp8266/snprintf.c [hunk 9]
```diff
@@ -811,7 +604,9 @@ int main (void)
 }
 #endif /* SNPRINTF_TEST */
 
-void abort(void) {}
+void abort(void) {
+    while (1);
+}
 
 #endif /* !HAVE_SNPRINTF */
 /* Shut up the compaq compiler which hates empty files. This will
```
Comment (RostakaGmfun) [inline client/client-multi/client-c/src/kaa/platform-impl/esp8266/snprintf.c:612]: Are we going to use that mystical compiler?

Comment (rasendubi) [inline client/client-multi/client-c/src/kaa/platform-impl/esp8266/snprintf.c:612]: That will be handled outside of the scope of the ticket.

File: client/client-multi/client-c/build.sh (hunk 1 above)
Comment (rasendubi) [inline client/client-multi/client-c/build.sh:41]: KAA-858 is related but won't address Max's comment

Comment (rasendubi) [inline client/client-multi/client-c/build.sh:41]: But yeah! We should merge it to `true_metal` anyway

File: client/client-multi/client-c/src/kaa/platform-impl/cc32xx/logger.c (hunk 8 above)
Comment (rasendubi) [inline client/client-multi/client-c/src/kaa/platform-impl/cc32xx/logger.c:28]: Not checking `sink` will change the behavior and won't be refactoring anymore

**Comment by rasendubi** [comment]
@forGGe, @RostakaGmfun, most comments have been addressed. Re-review, please

**Comment by RostakaGmfun** [comment]
LGTM

File: client/client-multi/client-c/src/kaa/platform-impl/cc32xx/logger.c (hunk 8 above)
Comment (RostakaGmfun) [inline client/client-multi/client-c/src/kaa/platform-impl/cc32xx/logger.c:28]: If so, then such check should be added to `esp8266` implementation too.

Comment (rasendubi) [inline client/client-multi/client-c/src/kaa/platform-impl/cc32xx/logger.c:28]: That may not be necessary -- that may change nothing. But feel free to file a ticket if you think it does.

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c (hunk 6 above)
Comment (forGGe) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c:540]: Ah, I see. Do not bother then.

**Comment by forGGe** [comment]
LGTM

**Pull request merged**